test/TestProjects/NameConflicts/Generated/AutoRestParameterFlatteningRestClient.cs

```diff
@@ -233,9 +233,7 @@ internal Azure.Core.HttpMessage CreateAnalyzeBodyRequest(string stringBody)
             if (stringBody != null)
             {
                 request.Headers.Add("Content-Type", "application/json");
-                var content = new Utf8JsonRequestContent();
-                content.JsonWriter.WriteStringValue(stringBody);
-                request.Content = content;
+                request.Content = new StringRequestContent(stringBody);
```

Comment:
What made this change?

Reply:
The request content type is decided by Request.protocol.http.knownMediaType in M4. 
Previously, this request has protocol:
```
protocol: !Protocols 
  http: !HttpWithBodyRequest 
    path: /conflictingLROOverloads
    method: post
    knownMediaType: json
    mediaTypes:
      - application/json
    uri: '{$host}'
```
After m4 change, the protocol changes to:
```
protocol: !Protocols 
  http: !HttpBinaryRequest 
    path: /conflictingLROOverloads
    method: post
    binary: true
    knownMediaType: text
    mediaTypes:
      - application/json
    uri: '{$host}'
```
It looks like a regression. If you think so, I'll create an issue to Tim.

Reply:
So I have a fix for this case here, please give it a try and confirm it is resolving the issue https://github.com/Azure/autorest/pull/4506

However this bring another case that is not handle perfection due to the fact that `knownMediaType` is the same regardless of the content type passed.

In the case you have 
```yaml
consumes:
- application/json
- text/plain
- application/pdf
parameters: 
  - in: body
    schema:
      type: string
```

This would produce 2 overload

```
function foo(contentType: "application/pdf", bytes: Stream);
function foo(contentType: "application/json" | "text/plain", content: string);
```

In theory if you pass `application/json` as the content type for the other overload it should the content as a json string but if you pass text/plain it should pass it raw.

This means we should really need a map from content-Type to knownMediaType here which can be used to serialize differently depending on the passed type.

Reply:
@timotheeguerin Where do you think we should store this map?

Reply:
@annelo-msft I tried this scenario, if we have type 
- application/json
- text/plain
Then the content will be `StringRequestContent`, which is only for `text/plain`.
```C#
internal Azure.Core.HttpMessage CreateAnalyzeBodyRequest(Models.ContentType contentType, string stringBody)
{
    var message = _pipeline.CreateMessage();
    var request = message.Request;
    request.Method = RequestMethod.Post;
    var uri = new RawRequestUriBuilder();
    uri.Reset(_endpoint);
    uri.AppendPath("/conflictingLROOverloads", false);
    request.Uri = uri;
    if (stringBody != null)
    {
        request.Headers.Add("Content-Type", contentType.ToSerialString());
        request.Content = new StringRequestContent(stringBody);
    }
    return message;
}
```
We might need to change the implementation to 
```C#
internal Azure.Core.HttpMessage CreateAnalyzeBodyRequest(Models.ContentType contentType, string stringBody)
{
    var message = _pipeline.CreateMessage();
    var request = message.Request;
    request.Method = RequestMethod.Post;
    var uri = new RawRequestUriBuilder();
    uri.Reset(_endpoint);
    uri.AppendPath("/conflictingLROOverloads", false);
    request.Uri = uri;
    if (stringBody != null)
    {
        if (contentType == Models.ContentType.TextPlain)
        {
            request.Headers.Add("Content-Type", contentType.ToSerialString());
            request.Content = new StringRequestContent(stringBody);
        }
        else if (contentType == Models.ContentType.ApplicationJson)
        {
            request.Headers.Add("Content-Type", "application/json");
            var content = new Utf8JsonRequestContent();
            content.JsonWriter.WriteStringValue(stringBody);
            request.Content = content;
        }
    }
    return message;
}
```